vms/platformvm/state/state.go

```diff
@@ -1650,12 +1823,20 @@ func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error
 				staker := validatorDiff.validator
 				weightDiff.Amount = staker.Weight
 
-				// Invariant: Only the Primary Network contains non-nil
-				//            public keys.
+				// Invariant: Only the Primary Network contains non-nil public
+				// keys.
 				if staker.PublicKey != nil {
 					// Record the public key of the validator being removed.
-					pkDiffs[nodeID] = staker.PublicKey
+					err := s.flatValidatorPublicKeyDiffsDB.Put(
+						getDiffKey(constants.PrimaryNetworkID, height, nodeID),
+						staker.PublicKey.Serialize(),
```

Comment:
Shouldn't this be `nil` and the one above be the `pk`?

Reply:
Or, do we record the previous entry?

Reply:
No - these are reverse diffs.

Reply:
Much better.